Fixed the students locating logic

almogcarmi opened the pull request.

almogcarmi committed:
Fixed the students locating logic to make sure that only one students fits the received name.

almogcarmi committed:
Fixed the students locating logic to make sure that only one students fits the received name.

kobybum committed:
Sections: Run black on code.

kobybum committed:
Assignments: Avoid re-fetching resources for each section.

Previously all the information was fetched from moodle for each section.
Instead we fetch info once and then regroup it after.

maxandron commented on moodler/students.py:
```diff
@@ -21,31 +23,41 @@ def core_enrol_get_enrolled_users(course_id):
 
 
-def get_students(course_id, students_names=None):
+def get_students(course_id, students_names: list = None):
```
I'm having trouble understanding this function in this context...
Document the parameters please. What is student_names? When would this be None?
Also, students_names_dict is basically a dict of student_name to found_status? If yes, change the name to be more descriptive, maybe add a comment. students_names_dict is not informative enough...

almogcarmi replied:
You are totally right. I have decided that I was trying to enforce two functions to be one function.
I created a whole new function to be used for retrieving students IDs by their names.

maxandron commented on moodler/students.py:
```diff
@@ -21,31 +23,41 @@ def core_enrol_get_enrolled_users(course_id):
 
 
-def is_student_in_names(students_names, enrolled):
+def is_student_in_names(students_names_dict, enrolled):
```
So this function always returns False, no?

maxandron commented on moodler/students.py:
```diff
@@ -21,31 +23,41 @@ def core_enrol_get_enrolled_users(course_id):
-        if students_names is not None:
-            if not is_student_in_names(students_names, enrolled):
+        if students_names_dict is not None:
+            if not is_student_in_names(students_names_dict, enrolled):
```
Wouldn't this always continue?

kobybum committed:
Merge pull request #26 from DARTSG/feature/optimize-assignments

Optimize Assignment Fetching

almogcarmi committed:
Separated the logic of retrieving a list of all the students and retrieving a list of specific students.

1. Created a new function to be called for retrieving a list of specific
   students IDs.
2. Made sure that a student will only be found once - otherwise error!

almogcarmi committed:
Merge branch 'hotfix/fixed-students-name-finding' of https://github.com/DARTSG/moodler into hotfix/fixed-students-name-finding

almogcarmi committed:
Added calls to the submissions locking while validating that the students were found.

maxandron commented on moodler/sections.py:
```diff
@@ -75,13 +75,12 @@ def list_courses(course_prefix=None):
         if course_prefix is not None:
-            if ((course_prefix not in course['shortname']) and
-                    (course_prefix not in course['fullname'])):
+            if (course_prefix not in course["shortname"]) and (
```
```python
if not (course_prefix in course["shortname"] or course_prefix in course["full_name"])
```

maxandron replied:
actually something better

```python
if course_prefix not in (course["shortname"] + course["full_name"])
```

maxandron replied:
Unless `.starts_with` is more appropriate here

maxandron commented on moodler/sections.py:
```diff
@@ -134,76 +138,64 @@ def locate_course_name(course_id, course_prefix=None):
-            # the moodle
-            if module['modname'] != 'assign':
+            # Filter assignments by name
+            if assignments_names and not module["name"] in assignments_names:
```
`not "a" in "abc"`
is the same as `"a" not in "abc"`

almogcarmi committed:
Small fixes regarding the CRs from Ron
